File: src/main/java/hilfling/backend/hilfling/controllers/SecurityLevelController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.SecurityLevel;
import hilfling.backend.hilfling.service.SecurityLevelService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/security_levels")
public class SecurityLevelController extends GenericBaseControllerImplementation<SecurityLevel> {

    @Autowired
    private SecurityLevelService service;
    public SecurityLevelService  getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/SecurityLevelController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.SecurityLevel
import hilfling.backend.hilfling.repository.SecurityLevelRepository
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/security_levels")
class SecurityLevelController {
    val repository = SecurityLevelRepository()

    @GetMapping("/{id}")
    fun getSecurityLevelById(@PathVariable("id") id: Long): SecurityLevel? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAllSecurityLevels(): List<SecurityLevel> {
        return repository.findAll()
    }

    @PostMapping
    fun createSecurityLevel(@RequestParam("type") type: String): SecurityLevel {
        return repository.create(type)
    }
}